### PR TITLE
fmt: handy functions for pointer-length objects

### PR DESCRIPTION
**Pull request by cHuberCoffee**

### CODE REVIEW

File: src/fmt/pl.c [hunk 1]
```diff
@@ -355,6 +355,31 @@ int pl_bool(bool *val, const struct pl *pl)
...
+/**
+ * Convert an ASCII hex string as a pointer-length object to binary format
+ *
+ * @param pl  Pointer-length object
+ * @param hex Destination binary buffer
+ * @param len Length of binary buffer
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int pl_hex(const struct pl *pl, uint8_t *hex, unsigned int len)
```
Comment (alfredh) [inline src/fmt/pl.c:367]: please use `size_t` instead of unsigned int

File: src/fmt/pl.c [hunk 2]
```diff
@@ -355,6 +355,31 @@ int pl_bool(bool *val, const struct pl *pl)
...
+{
+	unsigned int i;
```
Comment (alfredh) [inline src/fmt/pl.c:369]: using C99 style, this variable can be declared inside the for-loop.

Also please use `size_t`

File: src/fmt/pl.c [hunk 3]
```diff
@@ -655,6 +680,39 @@ const char *pl_strrchr(const struct pl *pl, char c)
...
+	/*case str is empty or just '\0'*/
+	if (!str_len(str))
+		return pl->p;
+
+	for (i = 0; i < pl->l; ++i) {
+		/*case rest of pl is not long enough*/
+		if (pl->l - i < str_len(str))
+			return NULL;
+
+		if (!memcmp(pl->p + i, str, str_len(str)))
```
Comment (alfredh) [inline src/fmt/pl.c:708]: you can cache the length of the string, it should speed up the function slightly ...

File: src/fmt/pl.c [hunk 4]
```diff
@@ -655,6 +678,39 @@ const char *pl_strrchr(const struct pl *pl, char c)
 }
 
 
+/**
+ * Locate the fist substring in a pointer-length string
```
Comment (sreimers) [inline src/fmt/pl.c:682]: fist -> first

**Pull request merged**